src/stores/index.js

```diff
@@ -292,6 +348,9 @@ const Store = Reflux.createStore({
    * @param {Connection} connection - The favorite connection to copy.
    */
   onDuplicateConnectionClicked(connection) {
+    // Cancel any current connection attempt if there is one.
+    this._cancelCurrentConnectionAttempt();
```

Comment:
This could be a good topic for design sync, if it wasn't brought up already: are we sure we don't want to really stop the user from doing things while connecting? I think is unexpected for the connection to be interrupted as the side effect of .. quite anything that is not cancelling it intentionally i guess.